libnavigation-ui/src/main/java/com/mapbox/navigation/ui/route/NavigationMapRoute.java

```diff
@@ -341,6 +341,7 @@ public void showAlternativeRoutes(boolean alternativesVisible) {
    */
   public void addProgressChangeListener(MapboxNavigation navigation) {
     this.navigation = navigation;
+    mapRouteProgressChangeListener.updateVisibility(true);
```

Comment:
Talking offline, but maybe others have input so posting here. Calling update visibility here is side-effect

Potential interface issue is if I'm using NavigationMapRoute
``` kotlin
mapRoute.updateRouteVisibilityTo(false)
mapRoute.addProgressChangeListener(..)
``` 
^ route will be visible.

```kotlin
mapRoute.addProgressChangeListener(..)
mapRoute.updateRouteVisibilityTo(false)
```
^ route is not visible

This could be perceived as a bug or a misuse of the interface.

Reply:
Fair enough but to add to the conversation, why does a non-visual route progress listener have an isVisible flag?  The listener is never visible.  It seems to me the listener should always process the events it receives. This this specific case the listener can still make updates to the route line even if it's not visible. The changes just wouldn't be seen. The isVisible flag seems irrelevant.

Reply:
Think a more appropriate place to update the route visibility is in NavigationMapboxMap

```
NavigationMapboxMap
  public void drawRoute(@NonNull DirectionsRoute route) {
    mapRoute.updateRouteVisibilityTo(true);
    mapRoute.addRoute(route);
  }
```

Reply:
What @cafesilencio mentioned is right that "a non-visual listener holds a isVisible flag". It sounds weird to me as well.

But since this change will potentially lead to the bug as @kmadsen mentioned above. We would be better to avoid it now. So we won't have same confusion/question in future that "why register a listener needs to update visibility?".

I'm thinking since the issue #3001 is because we hide the routeLine when we call `NavigationMapboxMap`'s 
```java
  public void removeRoute() {
    mapRoute.updateRouteVisibilityTo(false);
    mapRoute.updateRouteArrowVisibilityTo(false);
  }
```
maybe we could
* call `showRoute()` within `NavigationMapboxMap`'s `drawRoute()` or
* add `showRoute()` in `NavigationMapRoute`'s `addRoutes()` after **routeLine.draw(directionsRoutes);**?

my 2cents.